pcl_utilities/src/tests/simple_test_concatenate_point_cloud.cpp

```diff
@@ -0,0 +1,181 @@
+#include "rcl_interfaces/msg/parameter_descriptor.hpp"
+
+#include "sensor_msgs/msg/point_cloud2.hpp"
+
+#include "pcl_utility_msgs/srv/pcl_concatenate_point_cloud.hpp"
+
+using pcl_utility_msgs::srv::PCLConcatenatePointCloud;
+using sensor_msgs::msg::PointCloud2;
+
+constexpr size_t MAX_POINT_CLOUDS = 100U;
+constexpr std::chrono::seconds MAX_WAIT_TIME {1U};
+
+class TestConcatenatePointCloudNode: public rclcpp::Node
+{
+private:
+    rclcpp::Client<PCLConcatenatePointCloud>::SharedPtr
+      concatenate_point_cloud_client_;
+    rclcpp::Publisher<PointCloud2>::SharedPtr output_publisher_;
+    std::deque<PointCloud2> history_;
+    std::string camera_topic_;
+    size_t num_point_clouds_;
+
+public:
+  TestConcatenatePointCloudNode()
+  : rclcpp::Node("simple_test_concatenate_point_cloud")
+  {
+    std::string client_topic = declare_parameter<std::string>("node_client_name");
+    camera_topic_ = declare_parameter<std::string>("point_cloud_topic");
+
+    // rclcpp uses int64_t internally, use explicit cast to size_t
+    rcl_interfaces::msg::ParameterDescriptor param_constraints;
+    rcl_interfaces::msg::IntegerRange integer_range;
+    integer_range.from_value = 0U;
+    integer_range.to_value = MAX_POINT_CLOUDS;
+    param_constraints.integer_range.push_back(integer_range);
+
+    num_point_clouds_ = static_cast<size_t>(
+      declare_parameter<int64_t>("num_point_clouds", std::move(param_constraints)));
+
+    concatenate_point_cloud_client_ =
+      create_client<PCLConcatenatePointCloud>(client_topic);
+
+    output_publisher_ = create_publisher<PointCloud2>(
+      "concatenate_point_cloud/cloud_concatenated", 1);
+
+    history_ = std::deque<PointCloud2>();
+  }
+
+  void spin()
+  {
+    while(rclcpp::ok())
+    {
+      PointCloud2 point_cloud_message;
+      bool was_retrieved = rclcpp::wait_for_message(
+        point_cloud_message, shared_from_this(),
+        camera_topic_, MAX_WAIT_TIME);
+
+      if (!was_retrieved)
+      {
+        RCLCPP_ERROR(get_logger(),
+          "A camera message could not be retrieved within 1 second.");
+        continue;
+      }
+
+      process_point_cloud(point_cloud_message);
+    }
+  }
+
+  void process_point_cloud(PointCloud2& point_cloud) {
+    // create a fixed sized context window of old point clouds
+    history_.push_back(std::move(point_cloud));
+    if (history_.size() > num_point_clouds_)
+    {
+      history_.pop_front();
+    } else if (history_.size() < num_point_clouds_)
+    {
+      return;
+    }
+
+    auto request = std::make_shared<PCLConcatenatePointCloud::Request>();
+    request->cloud_list_in = std::vector(history_.cbegin(), history_.cend());
+
+    auto response_future =
+      concatenate_point_cloud_client_->async_send_request(request);
+
+    auto response_code = rclcpp::spin_until_future_complete(
+      shared_from_this(), response_future, MAX_WAIT_TIME);
+
+    if (response_code != rclcpp::FutureReturnCode::SUCCESS)
+    {
```

Comment:
Brace should cuddle on if: https://docs.ros.org/en/humble/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#open-versus-cuddled-braces